numpyro/distributions/constraints.py

```diff
@@ -88,6 +88,14 @@ def __call__(self, value):
         return (value > self.lower_bound) & (value < self.upper_bound)
 
 
+class _Multinomial(Constraint):
+    def __init__(self, upper_bound):
+        self.upper_bound = upper_bound
+
+    def __call__(self, value):
+        return (value >= 0) & (np.sum(value, -1) == self.upper_bound)
```

Comment:
I think we should use `np.all(value >=0, axis=-1)` instead of `value >= 0` to return a correct shape.

Reply:
Oops..you are absolutely right, will change.